### ISSUE TITLE
Clipboard being updated by non-active screen

### ISSUE THREAD

**Issue by soraxas** [input-leap/input-leap#1433]
### What happened?

Would it be possible to add an option to disable clipboard sharing across devices?

The reason is that I think it currently conflicts with other clipboard sharing applications. I have been using [Join](https://joaoapps.com/join/) which include sharing clipboard across clients (across devices such as PC/laptops/android phone/etc.).

Currently while using barrier (or the to-be input-leap), using PC1 as host, copying and pasting on client (PC2) will somehow erase the content. The following are logs from Host (while operating on Client):
```
[2022-01-15T14:42:47] INFO: screen "Client" grabbed clipboard 0 from "Host"
[2022-01-15T14:42:47] INFO: screen "Client" grabbed clipboard 1 from "Host"
# I think at this point *Join* had sent a network request to update clipboard on PC1 (Host). I haven't done any other operations at this point
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 0 from "Client"
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 1 from "Client"
```
where the lines
```
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 0 from "Client"
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 1 from "Client"
```
ended up resetting the clipboard.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

Host: Linux by compiling
Client: Windows from release

### What OSes are you seeing the problem on? (Check all that apply)

Linux, Windows

### What OS versions are you using?

Host: Arch Linux with latest Kernel
Client: Windows 10

**Comment by albertony** [comment]
There is an option in GUI: Server Configuration > Advanced server settings > Enable clipboard sharing.
Is this not covering what you want? I have not tested it myself, so its an open question..

**Comment by soraxas** [comment]
Oh! I haven't even realised there are options there (I've only been looking at `Barrier > Settings`). Thanks for pointing that out. It currently works as what I had in my mind. Btw I recalled seeing some post about making Barrier server-client agnostic (keyboard/mouse works both way). When that happens, perhaps these settings should be moved to the general location?

However, that sparks my other question of shouldn't the clipboard grabbing only happens on the current active screen? Say, current barrier currently keep track of the **active** screen:
```
[2022-01-16T10:04:09] INFO: switch from "Client" to "Host" at 1887,659
[2022-01-16T10:04:09] INFO: entering screen
[2022-01-16T10:04:09] INFO: switch from "Host" to "Client" at 0,418
[2022-01-16T10:04:09] INFO: leaving screen
```
The clipboard should not change if it's not the currently active client having its clipboard updates in the following log
```
# Currently, Client is active:
[2022-01-15T14:42:47] INFO: screen "Client" grabbed clipboard 0 from "Host"
[2022-01-15T14:42:47] INFO: screen "Client" grabbed clipboard 1 from "Host"
# Host's clipboard was set by external program
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 0 from "Client"
[2022-01-15T14:42:50] INFO: screen "Host" grabbed clipboard 1 from "Client"
```

---------------

The above is reproducible by running the following in host:
```sh
sleep 5 && echo hello | xclip -in -selection clipboard
```
 and then switching screen to client immediately.

Even though the active screen (after 5 seconds) is client, the client's clipboard is forced to be updated when a background process has updated host's clipboard

**Comment by p12tic** [comment]
This issue has been addressed in #1434, closing.
